src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+ *
+ * It is passed to Core::SetProperty(), this option should be used with values:
+ * ov::runtime::config::value::yes or ov::runtime::config::value::no
+ */
+static constexpr Key<bool> enable_profiling{"PERF_COUNT"};
+
+namespace log {
+
+/**
+ * @brief Enum to define possible affinity template hints
+ */
+enum class Level {
+    NO = -1,      //!< disable any loging
+    ERR = 0,      //!< error events that might still allow the application to continue running
+    WARNING = 1,  //!< potentially harmful situations which may further lead to ERROR
+    INFO = 2,     //!< informational messages that display the progress of the application at coarse-grained level
+    DEBUG = 3,    //!< fine-grained events that are most useful to debug an application.
+    TRACE = 4,    //!< finer-grained informational events than the DEBUG
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const Level& level) {
+    switch (level) {
+    case Level::NO:
+        return os << "NO";
+    case Level::ERR:
+        return os << "LOG_ERROR";
+    case Level::WARNING:
+        return os << "LOG_WARNING";
+    case Level::INFO:
+        return os << "LOG_INFO";
+    case Level::DEBUG:
+        return os << "LOG_DEBUG";
+    case Level::TRACE:
+        return os << "LOG_TRACE";
+    default:
+        throw ov::Exception{"Unsupported log level"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, Level& level) {
+    std::string str;
+    is >> str;
+    if (str == "NO") {
+        level = Level::NO;
+    } else if (str == "LOG_ERROR") {
+        level = Level::ERR;
+    } else if (str == "LOG_WARNING") {
+        level = Level::WARNING;
+    } else if (str == "LOG_INFO") {
+        level = Level::INFO;
+    } else if (str == "LOG_DEBUG") {
+        level = Level::DEBUG;
+    } else if (str == "LOG_TRACE") {
+        level = Level::TRACE;
+    } else {
+        throw ov::Exception{"Unsupported log level: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief the key for setting desirable log level.
+ */
+static constexpr Key<Level> level{"LOG_LEVEL"};
+}  // namespace log
+
+/**
+ * @brief This key defines the directory which will be used to store any data cached by plugins.
+ *
+ * The underlying cache structure is not defined and might differ between OpenVINO releases
+ * Cached data might be platform / device specific and might be invalid after OpenVINO version change
+ * If this key is not specified or value is empty string, then caching is disabled.
+ * The key might enable caching for the plugin using the following code:
+ *
+ * @code
+ * ie.set_property("GPU", ov::cache_dir("cache/")); // enables cache for GPU plugin
+ * @endcode
+ *
+ * The following code enables caching of compiled network blobs for devices where import/export is supported
+ *
+ * @code
+ * ie.set_property(ov::cache_dir("cache/")); // enables models cache
+ * @endcode
+ */
+static constexpr Key<std::string> cache_dir{"CACHE_DIR"};
```

Comment:
Why we use `log::level`, but `cache_dir` ? Shouldn't it be aligned to either `log::level`, `cache::dir` or `log_level`, `cache_dir`?

Reply:
I suppose log_level has also possible values which are in the same namespace.

Reply:
log namespace contains some other log-specific details like `log::Level::` enum values. `cach_dir` is just `cahe_dir`